server/src/main/kotlin/com/xebia/functional/xef/server/http/routes/Requests.kt

```diff
@@ -0,0 +1,38 @@
+package com.xebia.functional.xef.server.http.routes
+
+import io.ktor.resources.*
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class ChatMessage(
+    val role: ChatRole, val content: String? = null, val name: String? = null
+)
+
+@JvmInline
+@Serializable
+value class ChatRole(val role: String) {
+    public companion object {
+        public val System: ChatRole = ChatRole("system")
+        public val User: ChatRole = ChatRole("user")
+        public val Assistant: ChatRole = ChatRole("assistant")
+        public val Function: ChatRole = ChatRole("function")
+    }
+}
+
+
+@Serializable
+@Resource("/chat/completions")
+data class PromptMessageRequest(
+    val model: String,
+    val messages: List<ChatMessage>,
+    val temperature: Double = 1.0,
```

Comment:
Are we sure that we want `temperature` =  `1.0` by default?

Reply:
Yes, it's better not to have this value. But now this class doesn't exist, so problem solved 🤓